### DISCUSSION TITLE
Material property 'matpp', requested by 'max_stress_excess' is not defined on

### DISCUSSION THREAD

**Discussion by Rahim-Habibi** [idaholab/moose#27267] [Q&A General] [answered]
Hi all,

in my simulation I need to track a material property by which to stop/start a step e.g. stoping injection or production from a geothermal reservoir. To track the material property, I have tried to use "ElementExtremeMaterialProperty" like the following but i faced up with an error. 
here is the a part of input file:

[Materials]
  [failure]
      type = TigerFailure
      block = 'fault_left fault_right fault_mid fault_bottom'
      property_name = 'matpp'
      outputs = exodus
      []
[]
[Postprocessors]
  [./max_stress_excess]
    type = ElementExtremeMaterialProperty
    mat_prop = matpp
    value_type = max
    execute_on = ' timestep_end'
    block = 'fault_left fault_right fault_mid fault_bottom'
  [../]
[]

and here is the error:
Material property 'matpp', requested by 'max_stress_excess' is not defined on fault_left fault_right fault_mid fault_bottom

Thanks in advance
Rahim

**Comment by GiudGiud** [comment]
Hello

Are these locations for the postprocessor sidesets?
I think you may need to pass an extra parameter to get material properties defined on sidesets 
Can you please check the Material main documentation page?

guillaume

**Reply by Rahim-Habibi** [reply]
Thanks Guillaume, 
They are not sidesets. Actually they are 2D elements attached to sidesets. 
what parameter has to be passed? I looked for it here: https://mooseframework.inl.gov/syntax/Materials/index.html , but could not find. 

Rahim

**Reply by GiudGiud** [reply]
Can you use Debug/show_material_props to check that the names of the properties declared and used match?

And check the block restrictions

**Reply by Rahim-Habibi** [reply]
I just add debug block in input file. The following shows the debug/show_material-props for "fault_bottom". I got the same for other subdomins i.e. fault_left fault_right fault_mid.

![image](https://github.com/idaholab/moose/assets/57484835/b682052d-32e0-4ef4-be8a-a009905a41a6)

the parameter (Coulomb_Stress) I want to track on shows up here. but I got the error.

**Reply by echo1115** [reply, marked as answer]
It is noted that `TigerFailure` is a custom material class created by you. In its source file (TigerFailure.C ), you should have already computed and declared some properties you need, taking `diffusivity `and `density `as examples：
```
_diffusivity(declareProperty<Real>("diffusivity")),
_density(declareProperty<Real>("density")),
```
Then if you want to track the maximum value of `diffusivity`, you can use this:
```
[Materials]
  [failure]
    type = TigerFailure
    block = 'fault_left fault_right fault_mid fault_bottom'
    #property_name = 'matpp'   #comment out this line
    #outputs = exodus         #comment out this line
  []
[]

[Postprocessors]
  [./max_stress_excess]
      type = ElementExtremeMaterialProperty
      mat_prop = diffusivity    # the property you declare in the source file
      value_type = max
      execute_on = ' timestep_end'
      block = 'fault_left fault_right fault_mid fault_bottom'
  [../]
[]
```

**Reply by GiudGiud** [reply]
which material is supposed to be declaring `matpp` ? I dont see it in the Debug log.
Could you please paste it as text next time, makes searching easier. With triple back-quotes, the formatting of text is fairly decent

**Reply by Rahim-Habibi** [reply]
Thanks @echo1115 
It works in this way. But i did not expect it in this way because here (https://mooseframework.inl.gov/source/postprocessors/ElementExtremeMaterialProperty.html), it is mentioned to use "property_name". I followed the examples there.  
Any way, thanks. 
@GiudGiud you can close the discussion, the error did not show up after applying @echo1115 suggestion. 
Rahim
